scripts/wmt/README.md

```diff
@@ -1,49 +1,115 @@
+### Training the transformer model
 
-### Training
+Now that the train/valid/test files have been generated, kick off a training run using the following command. We recommend training on 4 GPUs to get the best performance.
 
-We recommend training on 4 GPUs to get the best performance:
 
 ```bash
 CUDA_VISIBLE_DEVICES=0,1,2,3 ./run_wmt_ende.sh
 ```
 
+If you have e.g. 3 GPUs, change the `--num-gpus=4` in the `run_wmt_ende.sh` script to `--num-gpus=3` and set `CUDA_VISIBLE_DEVICES=0,1,2`.
+
 Or if you have only 1 GPU, run the dedicated script:
 
 ```bash
 CUDA_VISIBLE_DEVICES=0 ./run_wmt_ende_1gpu.sh
 ```
 
-### Translate
+The training logs printed to STDOUT include the values:
++ `words_per_sec/features`: number of input words processed per second (where features = words in the training set)
++ `words_per_sec/labels`: number of output words processed per second (where labels = correct words in the target language train file)
++ `global_step/sec`: number of gradient updates/# batches processed per second. `global step` is a Tensorflow variable that counts the number of batches seen by the model.
+
+The reported values are the averages since the last printed output (not a rolling average).
+
+### Monitoring model training using Tensorboard
+
+You can launch Tensorboard training monitoring by specifying where the logs are being written to (in this case, to a folder called `wmt_ende_transformer`). For example:
+
+`tensorboard --logdir="/path/to/wmt_ende_transformer" --port=6006`
+
+You can then open a browser and go to `<IP address>:6006` (e.g. http://127.0.0.1:6006/ if running on your local machine) to see Tensorboard graphs of gradients, loss, BLEU, learning rate, etc. over time as training proceeds. For an introduction to using Tensorboard, see this [video](https://www.youtube.com/watch?v=eBbEDRsCmv4) or this [post](https://www.datacamp.com/community/tutorials/tensorboard-tutorial).
+
+If you are using NVIDIA GPUs, you can monitor card usage during training using `watch -n 0.5 nvidia-smi`.
+
+If you are using a Docker container, you can launch a Tensorboard run from outside the container using:
+
+`docker exec <CONTAINER ID> tensorboard --logdir="path/to/wmt_ende_transformer" --port=6006`
+
+Note that for this to work, you need to have exposed the port when you create your Docker container ([simple example](https://briancaffey.github.io/2017/11/20/using-tensorflow-and-tensor-board-with-docker.html)).
+
+### Translation using a trained model 
+
+You can run the following script to perform inference on the test set using a trained model. The script calls `onmt-main infer`. Normally, the latest checkpoint is used for inference by default, but [we recommend averaging the parameters of several checkpoints](http://opennmt.net/OpenNMT-tf/inference.html#checkpoints-averaging), which usually boosts model performance.
```

Comment:
You should mention that the training does that automatically if run until the end.